### PR TITLE
external: support merge data files on S3

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #45719

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: br/pkg/lightning/backend/external/merge.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package external
```
Comment (okJiang) [inline br/pkg/lightning/backend/external/merge.go:1]: lisense

Comment (lance6716) [reply br/pkg/lightning/backend/external/merge.go:1]: will fix it in next PR

**Pull request merged**